rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+    """ Convert 1D photon events into 2D image
+    Opposite of image_to_photon_events
+
+    Parameters
+    -----------
+    photon_events : np.array
+        three column x, y, Iph photon locations and intensities
+    bins : int or array_like or [int, int] or [array, array]
+        The bin specification in y then x order:
+            * If int, the number of bins for the two dimensions (nx=ny=bins).
+            * If array_like, the bin edges for the two dimensions
+              (y_edges=x_edges=bins).
+            * If [int, int], the number of bins in each dimension
+              (ny, nx = bins).
+            * If [array, array], the bin edges in each dimension
+              (y_edges, x_edges = bins).
+            * A combination [int, array] or [array, int], where int
+              is the number of bins and array is the bin edges.
+    Returns
+    -----------
+    x_centers : array
+        1D vector describing column position
+    y_centers : array
+        1D vector describing row position
+    image : array
+        2D image
+    x_centers : array
+        1D vector describing bin edges for columns
+    y_centers : array
+        1D vector describing row position
+    """
+    x = photon_events[:, 0]
+    y = photon_events[:, 1]
+    Iph = photon_events[:, 2]
+
+    if bins is None:
+        ybins = np.arange(y.min()//1 + 0.5, y.max()//1, 1)
+        xbins = np.arange(x.min()//1 + 0.5, x.max()//1, 1)
+        bins = (ybins, xbins)
+
+    image, y_edges, x_edges = np.histogram2d(y, x, bins=bins, weights=Iph)
+    y_centers = (y_edges[:-1] + y_edges[1:])/2
+    x_centers = (x_edges[:-1] + x_edges[1:])/2
+
+    return x_centers, y_centers, image, x_edges, y_edges
+
+
+def run_test():
+    """Run at test of the code.
+    This can also be used as a reference for command-line analysis.
+
+    Returns
+    -------------
+    ax1 : matplotlib axis object
+        axes used to plot the image
+    ax2 : matplotlib axis object
+        axes used to plot the spectrum
+    """
+    photon_events = make_fake_image()
+    curvature = fit_curvature(photon_events)
+    print("Curvature is\n{} \t {:.3e}\n{}\t{:.3e}\n"
+          "{}\t{:.3e}".format(*sum(curvature.items(), ())))
+
+    fig1, ax1 = plt.subplots()
+    plt.title('Image')
+    fig2, ax2 = plt.subplots()
+    plt.title('Spectrum')
+
+    plot_scatter(ax1, photon_events)
+    plot_curvature(ax1, curvature, photon_events)
+
+    spectrum = extract(photon_events, curvature)
+    ax2.plot(spectrum[:, 0], spectrum[:, 0], '.')
+    # Depreated function will be added back later
+    # resolution = fit_resolution(spectrum)
+    # plot_resolution(ax2, spectrum)
+    # plot_resolution_fit(ax2, spectrum, resolution)
+    # print("Resolution is {}".format(resolution[0]))
+
+    return ax1, ax2
+
+
+if __name__ == "__main__":
+    """Run test of code"""
+    print('Run a test of the code')
+    run_test()
+
+# DEPRECIATED FUNCTIONS HERE TO BE UPGRADED LATER
```

Comment:
All of these I think should be removed, add 'issues 'for things that need to be added later or add another PR with the label 'DO NOT MERGE' and explain that they are unfinished.Or better yet create your own branch to keep these in until they are completed

Reply:
Ok